packages/typed/src/generate-schema.ts

```diff
@@ -64,13 +64,19 @@ const transformPropertySignature = <T extends ts.Node>(
     if (!ts.isPropertySignature(node)) {
       return node;
     }
-    const ps = node as ts.PropertySignature;
+    const ps = node;
     if (!ps.type || !ts.isTypeReferenceNode(ps.type)) {
       return node;
     }
-    const tr = ps.type as ts.TypeReferenceNode;
+    const tr = ps.type;
     if ((tr.typeName as any).escapedText === "Placeholder") {
-      ps.type = tr.typeArguments[0];
+      return ts.factory.updatePropertySignature(
```

Comment:
In TypeScript visitor, AST nodes are immutable and we should use `ts.factory.updateXxxx` if we want touch them.